### DISCUSSION TITLE
Able to create duplicate edge with multiplicity SIMPLE

### DISCUSSION THREAD

**Discussion by krittamk** [JanusGraph/janusgraph#2620] [Q&A]
As per my understanding of janusgraph [docs][1] there can only be at most one edge of multiplicity SIMPLE between two vertices.

I'm trying to represent a 'friend' relationship here i.e. if A is a friend of B, B is also a friend of A (undirected relationship).

But in practice I've been able to create two edges between A and B

> A ---->B
> B ---->A

    m = graph.openManagement()
    gremlin> m.getEdgeLabel("friend").multiplicity()
    ==>SIMPLE
    
    v1 = g.addV("person").property("id","qazwsx").next()
    ==>v[1724552]
    gremlin> v2 = g.addV("person").property("id","xswzaq").next()
    ==>v[1728648]
    gremlin> e1 = g.V(v1).addE("friend").to(v2).next()
    ==>e[ivwx-10yo8-1lh-111u0][1724552-friend->1728648]
    gremlin> e2 = g.V(v2).addE("friend").to(v1).next()
    ==>e[iwb5-111u0-1lh-10yo8][1728648-friend->1724552]

Please resolve this confusion. 
In any case how do i enforce such a schema ?

  [1]: https://docs.janusgraph.org/basics/schema/

**Comment by vtslab** [comment]
This is intended behavior, but the wording in the reference documents is ambiguous. So, pair (v1, v2) != pair (v2, v1). You can also check in the source code that the uniqueness check works this way (line716):
[https://github.com/JanusGraph/janusgraph/blob/v0.5.3/janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java](https://github.com/JanusGraph/janusgraph/blob/v0.5.3/janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java)
Also note that in a distributed system one could not prevent two independent transactions to create edges in both directions concurrently, because there is nothing yet than can be locked (see [ref docs on locking](https://docs.janusgraph.org/advanced-topics/eventual-consistency/#data-consistency)).
